### ISSUE TITLE
[OptiFine Parity] Armor trim components fail to match correctly

### ISSUE THREAD

**Issue by Thanos1716** [SHsuperCM/CITResewn#466]
**The Issue**
When trying to match armor trim data which is stored in the component `minecraft:trim`, it is impossible to get it to match. This is in contrast to OptiFine which is able to match it without issue. I am not sure what is causing the discrepancy.

My `.properties` files look something like the following:

```properties
type=item
matchItems=minecraft:chainmail_boots
model=minecraft:item/boots/chainmail/coast/amethyst
components.trim.pattern=minecraft:coast
components.trim.material=minecraft:amethyst
```

**To Reproduce**
1. Install [this](https://github.com/user-attachments/files/17069510/visual_armor_trims.zip) development build of the [Visual Armor Trims](https://modrinth.com/resourcepack/visual-armor-trims) resource pack in OptiFine.
2. Create any piece of armor with a trim (avoid Bolt or Flow, as they are still being tested).
3. Observe the missing model. *Note: it is shown as a missing model in OptiFine because OptiFine is unable to handle the vanilla feature `paletted_permutations` with CIT models. CIT Resewn can handle them perfectly fine.*
4. Install the same development build in CIT Resewn.
5. Observe no change in model or texture

[latest.log](https://github.com/user-attachments/files/17069606/latest.log)

**Checklist**
- [x] I have attached the `latest.log` file relevant to this issue
- [x] I have checked the issues and have not found an existing case of this issue
- [x] I am using the latest version of CIT Resewn for my version of minecraft
- [x] I am using the mod on Fabric Loader
- [x] I have disabled broken paths in the mod's config and the issue persists

**Comment by SHsuperCM** [comment]
Same problem as #465, trims are in a registry and it couldnt match against it.

**Comment by SHsuperCM** [comment]
Fixed in e009f189 and will be out in the next beta (v1.2.1)
